roles/core/dns_server/tasks/main.yml

```diff
@@ -90,13 +59,60 @@
   tags:
     - template
 
-- name: include_tasks ░ Configure Master DNS
-  include_tasks: master.yml
-  when: "dns_master in (network_interfaces | map(attribute='ip4') | list)"
+- name: template █ Generate Forward zone file on Master DNS /var/named/forward
+  template:
+    src: forward.j2
+    dest: /var/named/forward
+    owner: root
+    group: root
+    mode: 0644
+  register: zoneconf
+  tags:
+    - template
+
+- name: set_fact ░ Set SOA Serial for reverse zone
+  set_fact:
+    serial: "{{ lookup('pipe', 'date +%s') }}"
+  when: zoneconf.changed  # noqa 503
   tags:
     - template
     - internal
 
+- name: template █ Generate SOA for forward zone file /var/named/forward.soa
+  template:
+    src: forward.soa.j2
+    dest: /var/named/forward.soa
+    owner: root
+    group: root
+    mode: 0644
+  notify: service █ Restart dns services
+  when: zoneconf.changed  # noqa 503
+  tags:
+    - template
+
+- name: template █ Generate Reverse zone file on Master DNS /var/named/reverse
+  template:
+    src: reverse.j2
+    dest: /var/named/reverse
+    owner: root
+    group: root
+    mode: 0644
+  register: zoneconf
+  tags:
+    - template
+
```

Comment:
You must repeat the "set_fact ░ Set SOA Serial for forward zone" here. The one at line 73 can be skipped if there is no change to forward zone, which would led to an undefined `serial` and a failure to template reverse.soa.

Reply:
I was thinking about this, and in fact both reverse and forward uses the same variables, so if a change there is in one, a change there is in the other. This is why I removed the second one.

Reply:
If we assume a change of the forward file *always* implies a change of the reverse file, you can remove the `register: zoneconf` line 100 and rely on the `zoneconf` already registered at line 69. It will be clearer for future code readers.

Reply:
Done